code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AutoCarouselTemplateNotificationBuilder.java

```diff
@@ -0,0 +1,226 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+package com.adobe.marketing.mobile;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.Build;
+import android.widget.RemoteViews;
+import androidx.core.app.NotificationCompat;
+import com.adobe.marketing.mobile.campaignclassic.R;
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.services.ServiceProvider;
+import com.adobe.marketing.mobile.services.caching.CacheResult;
+import com.adobe.marketing.mobile.services.caching.CacheService;
+import com.adobe.marketing.mobile.util.StringUtils;
+import com.adobe.marketing.mobile.util.UrlUtils;
+import com.google.firebase.components.MissingDependencyException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+
+public class AutoCarouselTemplateNotificationBuilder {
+    private static final String SELF_TAG = "AutoCarouselTemplateNotificationBuilder";
+
+    static NotificationCompat.Builder construct(
+            final CarouselPushTemplate pushTemplate,
+            final Context context,
+            final String channelId,
+            final String packageName)
+            throws MissingDependencyException {
+        final RemoteViews smallLayout =
+                new RemoteViews(context.getPackageName(), R.layout.push_template_collapsed);
+        final RemoteViews expandedLayout =
+                new RemoteViews(context.getPackageName(), R.layout.push_template_auto_carousel);
+        final CacheService cacheService = ServiceProvider.getInstance().getCacheService();
+        final String cacheLocation = CampaignPushUtils.getAssetCacheLocation();
+
+        if (cacheService == null) {
+            throw new MissingDependencyException(
+                    "Cache service is null, notification will not be constructed.");
+        }
```

Comment:
Let us use our own custom exception instead of one from Firebase. Alternatively we can also use IllegalStateException.